src/AppRoot.tsx

```diff
@@ -129,6 +129,11 @@ const initialState: State = {
 
 let checkCardInterval = 0
 
+let lastVoteUpdate = 0
+let lastVoteSaveToCard = 0
+let cardWriteInterval = 0
+let writingVoteToCard = false
```

Comment:
These should all be private instance variables on `AppRoot`, IMO. Otherwise each instance will share these values. This happens to not be a problem because–in testing–jest loads this class multiple times and resets the state, and in real use this component is never unmounted.

That said, it's just waiting to bite us when we do this somewhere that doesn't have this property. Can this functionality be split out into its own class? This file is getting awfully big.